src/protocol/encoder.js

```diff
@@ -173,15 +173,15 @@ module.exports = class Encoder {
   }
 
   writeEncoder(value) {
-    if (value instanceof Encoder !== true) {
+    if (value == null || value.buffer == null) {
```

Comment:
Nitpicking: `==` should be `===`; also below in `writeEncoderArray`.

Reply:
No, it's intentional to check for both null and undefined.